agimus_controller_ros/agimus_controller_ros/simple_trajectory_publisher.py

```diff
@@ -124,15 +104,14 @@ def publish_mpc_input(self):
 
         if self.pin_model is None:
             self.load_models()
-
-        self.scale_amp = self.quintic_trajectory(self.t)
-        amp = self.scale_amp * self.amp
         # Currently not changing the last two joints - fingers
         # for i in range(self.pin_model.nq - 2):
         for i in [2, 3]:
-            self.q[i] = self.q0[i] + amp * np.sin(self.w * self.t)
-            self.dq[i] = amp * self.w * np.cos(self.w * self.t)
-            self.ddq[i] = -amp * self.w * self.w * np.sin(self.w * self.t)
+            self.q[i] = (
+                self.q0[i] - self.amp + self.amp * np.sin(self.w * self.t + self.w)
```

Comment:
ok but then you still start from a non 0 acceleration?
at t = 0:
- q = q0 - A + A sin(pi/2) = q0 
- dq = A w cos(pi/2) = 0
- ddq = - A w w sin(pi/2) = - A w w

Reply:
Yes, you have to revised the formula. Either using a polynomial for the amplitude as you initially, or, what I would have done at first, using an exponential as

$`q = q0 + A (1 - \exp(-\alpha t) * \cos(\omega t)`$

which is easy to differentiate if you think of it as:

$`q = q0 + A \mathfrak{Real}\left(e^{i \omega t} - e^{ (- \alpha + i \omega) t} \right)`$